diff --git a/src/MarkdownTextInput.tsx b/src/MarkdownTextInput.tsx
index 82ee8953e823453998163a6646a4e233f8e7e1f6..8057b02963a13684ac7227bd8a57e01d4a32d42d 100644
--- a/src/MarkdownTextInput.tsx
+++ b/src/MarkdownTextInput.tsx
@@ -114,10 +114,12 @@ const MarkdownTextInput = React.forwardRef<MarkdownTextInput, MarkdownTextInputP
     return () => unregisterParser(parserId);
   }, [parserId]);
 
+  const plainMarkdownStyle = JSON.parse(JSON.stringify(markdownStyle));
+
   return (
     <MarkdownTextInputDecoratorViewNativeComponent
       style={styles.displayContents}
-      markdownStyle={markdownStyle}
+      markdownStyle={plainMarkdownStyle}
       parserId={parserId}
     >
       <TextInput
